packages/ldap/src/index.ts

```diff
@@ -0,0 +1,174 @@
+import { createServer, InvalidCredentialsError, NoSuchObjectError, OperationsError, Server } from 'ldapjs';
+import { LDAPOptions } from './types';
+import { SimulationState, Simulator } from '@simulacrum/server';
+import { ResourceServiceCreator } from '@simulacrum/server';
+import dedent from 'dedent';
+import { person } from '@simulacrum/server';
+import { spawn } from 'effection';
+import getPort from 'get-port';
+import { Slice } from '@effection/atom';
+import { assert } from 'assert-ts';
+
+const DefaultOptions: Partial<LDAPOptions> = {
+  port: 389
+};
+
+interface UserData{
+  id: string | number;
+  email: string;
+  password: string;
+}
+
+export function createLdapService<T extends UserData>(ldapOptions: LDAPOptions, state: Slice<SimulationState>): ResourceServiceCreator {
+  return () => {
+    let server: Server;
+
+    let getUsers = (): T[] => {
+      let users = state.slice('store', 'people').get();
+
+      assert(!!users, 'no scenarios in store/people');
+
+      return Object.values(users).map(u => ({ ...u, id: u.email })) as T[];
+    };
+
+    return {
+      name: 'ldap service',
+      *init() {
+        let port = ldapOptions.port ?? (yield getPort());
+        let baseDN = ldapOptions.baseDN;
+        let bindDn = ldapOptions.bindDn;
+        let bindPassword = ldapOptions.bindPassword;
+        let groupDN = ldapOptions.groupDN;
+
+        let log = {
+          debug: () => undefined,
+          trace: () => undefined,
+          warn: () => undefined,
+          error: () => undefined,
+        };
+
+        server = createServer({ log });
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        server.search(baseDN, function (req: any, res: any, next: any) {
+          console.log(dedent`--- User Search ---
+          dn:     ${req.dn.toString()}
+          scope:  ${req.scope}
+          filter: ${req.filter.toString()}
+          `);
```

Comment:
This should be behind a debug flag?

Reply:
we definitely could do that.

There is a real lack of good logging in simulacrum as a whole, I think we should have logging on by default while things are still taking shape.

This information is useful and you would use it as config for an application.

It can be really hard to find out why something is not working and I don't find much in the state, inspector or anywhere really so I end up logging stuff.

My shout would be to have more logging on by default and not less.